### PR TITLE
feat: add support for Casdoor authentication

### PR DESCRIPTION
**Pull request by jakiuncle**
## What is the purpose of the change

add support for Casdoor authentication

Use [Casdoor](https://github.com/casdoor/casdoor) for auth.

Features:

Support OIDC, OAuth 2.0, SAML, LDAP
With [Casbin](https://casbin.org/) based authorization management, Casdoor supports ACL, RBAC, ABAC, RESTful accessing control models
Front-end and back-end separate architecture, Casdoor supports high concurrency, provides web-based managing UI and have mature solution for springboot project https://casdoor.org/docs/integration/spring-boot
Casdoor supports Github, Google, QQ, WeChat third-party applications login, and support the extension of third-party login with plugins.
Phone verification code, email verification code and forget password features.
Accessing logs auditing and recording.
Casdoor supports integration with existing systems using db sync method, users can transition to Casdoor smoothly.
Casdoor supports mainstream databases: MySQL, PostgreSQL, SQL Server etc, and support the extension of new database with plugins.

### CODE REVIEW

**Comment by jakiuncle** [comment]
@qiaojialin @jixuan1989 @JulianFeinauer

File: backend/src/main/resources/application.properties [hunk 1]
```diff
@@ -18,4 +18,40 @@
...
 
 # Designate the configuration file
-spring.profiles.active=dev
\ No newline at end of file
+spring.profiles.active=dev
+
+casdoor.endpoint = http://localhost:7001
+casdoor.clientId = bba8e8be178cd1fb9b17
+casdoor.clientSecret = a0790543e686c5ce8bcaa6609005da14ac41a9a2
+casdoor.certificate =\
```
[the remark this answers is not in the document]
Comment (jakiuncle) [reply backend/src/main/resources/application.properties]: ok

File: frontend/src/i18n/en.js [hunk 2]
```diff
@@ -85,6 +85,7 @@ const en = {
       placeholderPassword: 'Please Input Password',
       forgetPassWord: 'Forget Password',
       signIn: 'Sign In',
+      signInWithCasdoor: 'sign In With Casdoor',
```
Comment (CritasWang) [inline frontend/src/i18n/en.js:88]: sign -> Sign

Comment (jakiuncle) [reply frontend/src/i18n/en.js]: ok

**Comment by jakiuncle** [comment]
@CritasWang plz review

**Pull request merged**